Fix warnings during the Mandyoc build

During the Mandyoc  build, there are some warnings ``-Wunused-but-set-variable``. Maybe we can fix. 

https://github.com/openjournals/joss-reviews/issues/3551

I got only a few variables marked `Wunused-but-set-variable`

I'll fix those.

> src/DM1_v.cpp: In function ‘PetscErrorCode AssembleA_Veloc(Mat, Mat, DM, DM)’:
> src/DM1_v.cpp:226:12: warning: variable ‘visc_meio’ set but not used [-Wunused-but-set-variable]
>   226 |  PetscReal visc_meio;
>       |            ^~~~~~~~~
> /opt/petsc/debug-v3.14/bin/mpicc -Wall -fdiagnostics-color -c src/DMT.cpp -o src/DMT.o -I/opt/petsc//include -I/opt/petsc//debug-v3.14/include
> /opt/petsc/debug-v3.14/bin/mpicc -Wall -fdiagnostics-color -c src/DMv.cpp -o src/DMv.o -I/opt/petsc//include -I/opt/petsc//debug-v3.14/include
> src/DMv.cpp: In function ‘PetscReal montaKeVeloc_simplif(PetscReal*, PetscReal*, PetscReal*)’:
> src/DMv.cpp:744:9: warning: variable ‘Hx’ set but not used [-Wunused-but-set-variable]
>   744 |  double Hx,Hz;
>       |         ^~
> src/DMv.cpp:744:12: warning: variable ‘Hz’ set but not used [-Wunused-but-set-variable]
>   744 |  double Hx,Hz;
>       |            ^~

--- a/src/DM1_v.cpp
+++ b/src/DM1_v.cpp
@@ -113,38 +113,38 @@ extern PetscReal theta_FSSA;
 extern PetscInt periodic_boundary;
 
 PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
-	
+
 	PetscErrorCode         ierr;
-	
+
 	PetscInt i,j,k;
-	
+
 	PetscInt               M,P;
-	
+
 	MatStencil indr[T_NE],ind1[1],ind2[1],ind[V_GT], indp[1];
-	
+
 	PetscScalar u[V_GT*V_GT],val_cond[1];
-	
+
 	Stokes					**pc;
-	
+
 	ierr = VecZeroEntries(local_Precon);CHKERRQ(ierr);
 	ierr = DMDAVecGetArray(veloc_da,local_Precon,&pc);CHKERRQ(ierr);
-	
+
 	Stokes					**VVC;
-	
+
 	ierr = VecZeroEntries(local_VC);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(veloc_da,Veloc_Cond,INSERT_VALUES,local_VC);
 	ierr = DMGlobalToLocalEnd(  veloc_da,Veloc_Cond,INSERT_VALUES,local_VC);
-	
+
 	ierr = DMDAVecGetArray(veloc_da,local_VC,&VVC);CHKERRQ(ierr);
-	
+
 	PetscInt       sx,sz,mmx,mmz;
-	
+
 	ierr = DMDAGetCorners(veloc_da,&sx,&sz,NULL,&mmx,&mmz,NULL);CHKERRQ(ierr);
-	
+
 	ierr = DMDAGetInfo(veloc_da,0,&M,&P,NULL,0,0,0, 0,0,0,0,0,0);CHKERRQ(ierr);
 
-	
+
 	for (k=sz; k<sz+mmz; k++) {
 		for (i=sx; i<sx+mmx; i++) {
 			if (periodic_boundary==1 && i==Nx-1){
@@ -153,7 +153,7 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 				ierr = MatSetValuesStencil(A,1,ind1,1,ind1,val_cond,ADD_VALUES);
 				ind1[0].c = 1;
 				ierr = MatSetValuesStencil(A,1,ind1,1,ind1,val_cond,ADD_VALUES);
-				
+
 				val_cond[0]=-1.0;
 				ind2[0].i = Nx;
 				ind2[0].j = k;
@@ -172,67 +172,66 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 				//ind1[0].i = i;  ind1[0].k = k; ind1[0].c = 1;
 				//val_cond[0] = 1.0-VVC[k][i].v;
 				//ierr = MatSetValuesStencil(A,1,ind1,1,ind1,val_cond,ADD_VALUES);
-				
+
 				ind1[0].i = i;  ind1[0].j = k; ind1[0].c = 1;
 				val_cond[0] = 1.0-VVC[k][i].w;
 				ierr = MatSetValuesStencil(A,1,ind1,1,ind1,val_cond,ADD_VALUES);
 			}
 		}
 	}
-	
+
 	int n,g;
-	
+
 	PetscInt               sex,sez,mx,mz;
 	PetscInt               ei,ek;
-	
+
 	ierr = DMDAGetElementCorners(veloc_da,&sex,&sez,&mx,&mz);CHKERRQ(ierr);
-	
-	
+
+
 	PetscScalar             **qq;
-	
+
 	ierr = VecZeroEntries(local_geoq);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(temper_da,geoq,INSERT_VALUES,local_geoq);
 	ierr = DMGlobalToLocalEnd(  temper_da,geoq,INSERT_VALUES,local_geoq);
-	
+
 	ierr = DMDAVecGetArray(temper_da,local_geoq,&qq);CHKERRQ(ierr);
 
 	PetscScalar 			**rr;
 
 	ierr = VecZeroEntries(local_dRho);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(temper_da,dRho,INSERT_VALUES,local_dRho);
 	ierr = DMGlobalToLocalEnd(  temper_da,dRho,INSERT_VALUES,local_dRho);
-	
+
 	ierr = DMDAVecGetArray(temper_da,local_dRho,&rr);CHKERRQ(ierr);
-	
-	
-	
-	
-	
-	
+
+
+
+
+
+
 	PetscReal volume = dx_const*dz_const;
-	
-	
+
+
 	PetscReal geoq_ele[T_NE],rho_ele[T_NE];
-	
+
 	visc_aux_MAX = 1.0E-50;
 	visc_aux_MIN = 1.0E50;
-	
-	
-	
-	
-	
-	PetscReal visc_meio;
+
+
+
+
+
 
 	PetscReal rho_mean_bottom;
 	PetscReal traction_bottom;
-	
+
 
 	for (ek = sez; ek < sez+mz; ek++) {
-		
+
 		for (ei = sex; ei < sex+mx; ei++) {
-			
+
 			/* check: future 3d version
 			indr[0].i=ei  ; indr[0].j=ej  ; indr[0].k=ek  ;
 			indr[1].i=ei+1; indr[1].j=ej  ; indr[1].k=ek  ;
@@ -243,18 +242,18 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 			indr[6].i=ei  ; indr[6].j=ej+1; indr[6].k=ek+1;
 			indr[7].i=ei+1; indr[7].j=ej+1; indr[7].k=ek+1;
 			 */
-			
+
 			indr[0].i=ei  ; indr[0].j=ek  ;
 			indr[1].i=ei+1; indr[1].j=ek  ;
 			indr[2].i=ei  ; indr[2].j=ek+1;
 			indr[3].i=ei+1; indr[3].j=ek+1;
-			
-			for (i=0;i<T_NE;i++) geoq_ele[i]=qq[indr[i].j][indr[i].i];
-			
 
-			visc_meio = montaKeVeloc_simplif(Ke_veloc,Ke_veloc_general,geoq_ele);
-			
-			
+			for (i=0;i<T_NE;i++) geoq_ele[i]=qq[indr[i].j][indr[i].i];
+
+
+			montaKeVeloc_simplif(Ke_veloc,Ke_veloc_general,geoq_ele);
+
+
 			for (i=0;i<V_GT*V_GT;i++) Ke_veloc_final[i]=Ke_veloc[i]*volume;
 
 
@@ -262,8 +261,8 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 
 			if (free_surface_stab==1){
 				for (i=0;i<T_NE;i++) rho_ele[i]=rr[indr[i].j][indr[i].i];
-				
-				rho_mean_bottom = (-rho_ele[0] - rho_ele[1] + rho_ele[2] + rho_ele[3])/2.0; 
+
+				rho_mean_bottom = (-rho_ele[0] - rho_ele[1] + rho_ele[2] + rho_ele[3])/2.0;
 				traction_bottom = theta_FSSA*dt_calor_sec*rho_mean_bottom*gravity*dx_const/2.0;
 
 				Ke_veloc_final[1*9] += traction_bottom;
@@ -279,19 +278,19 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 			if (periodic_boundary==1){
 				if (nni==Nx-1) nni=Nx;
 			}
-			
+
 			ind[n].i=ei  ; ind[n].j=ek  ; ind[n].c=0; n++;
 			ind[n].i=ei  ; ind[n].j=ek  ; ind[n].c=1; n++;
-			
+
 			ind[n].i=nni; ind[n].j=ek  ; ind[n].c=0; n++;
 			ind[n].i=nni; ind[n].j=ek  ; ind[n].c=1; n++;
-			
+
 			ind[n].i=ei  ; ind[n].j=ek+1; ind[n].c=0; n++;
 			ind[n].i=ei  ; ind[n].j=ek+1; ind[n].c=1; n++;
-			
+
 			ind[n].i=nni; ind[n].j=ek+1; ind[n].c=0; n++;
 			ind[n].i=nni; ind[n].j=ek+1; ind[n].c=1; n++;
-			
+
 			for (n=0;n<4;n++){
 				g = 2*n;
 				if (VVC[indr[n].j][indr[n].i].u==0){
@@ -301,7 +300,7 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 					for (j=0;j<V_GT;j++) u[g*V_GT+j]=Ke_veloc_final[g*V_GT+j];
 
 				}
-				
+
 				g = 2*n+1;
 				if (VVC[indr[n].j][indr[n].i].w==0){
 					for (j=0;j<V_GT;j++) u[g*V_GT+j]=0.0;
@@ -309,142 +308,142 @@ PetscErrorCode AssembleA_Veloc(Mat A,Mat AG,DM veloc_da, DM temper_da){
 				else {
 					for (j=0;j<V_GT;j++) u[g*V_GT+j]=Ke_veloc_final[g*V_GT+j];
 				}
-				
+
 			}
-			
+
 			ierr = MatSetValuesStencil(A,V_GT,ind,V_GT,ind,u,ADD_VALUES);
-			
-			
-			
-			
+
+
+
+
 			if (Verif_VG==0){
 				indp[0].i=ei  ; indp[0].j=ek  ; indp[0].c=0;
-				
+
 				ierr = MatSetValuesStencil(AG,V_GT,ind,1,indp,VCe,ADD_VALUES);
 			}
-			
-			
+
+
 			//preconditioner construction
 			for (n=0;n<V_GT;n++){
 				pc[ek][ei].u+=VCe[n]*VCe[n]/Ke_veloc_final[n*V_GT+n];
 			}
 		}
 	}
-	
+
 	ierr = MatAssemblyBegin(A,MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
 	ierr = MatAssemblyEnd(A,MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
-	
+
 	MatGetDiagonal(A,Adiag);
 	VecReciprocal(Adiag);
 	MatDiagonalSet(VB,Adiag,INSERT_VALUES);
-	
-	
+
+
 	if (Verif_VG==0){
 		ierr = MatAssemblyBegin(AG,MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
 		ierr = MatAssemblyEnd(AG,MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
-		
+
 		Verif_VG=1;
 	}
-	
-	
+
+
 	ierr = DMDAVecRestoreArray(veloc_da,local_Precon,&pc);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalBegin(veloc_da,local_Precon,ADD_VALUES,Precon);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalEnd(veloc_da,local_Precon,ADD_VALUES,Precon);CHKERRQ(ierr);
-	
-	
-	
+
+
+
 	ierr = DMDAVecRestoreArray(veloc_da,local_VC,&VVC);
 	ierr = DMDAVecRestoreArray(temper_da,local_geoq,&qq);CHKERRQ(ierr);
 	ierr = DMDAVecRestoreArray(temper_da,local_dRho,&rr);CHKERRQ(ierr);
-	
+
 	//printf("Visc_min = %lg, Visc_max = %lg\n",visc_aux_MIN,visc_aux_MAX);
-	
-	
-	
+
+
+
 	PetscFunctionReturn(0);
 }
 
 PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
-	
+
 	PetscScalar             **rr;
 	Stokes					**ff,**ffp,**VV;
-	
+
 	Stokes					**pp;
-	
-	
+
+
 	PetscInt               M,P;
 	PetscErrorCode         ierr;
-	
+
 	PetscScalar				Vfe[V_GT],Vfe_P[V_GT],dr[T_NE];
-	
+
 	PetscInt               sex,sez,mx,mz;
 	PetscInt               ei,ek;
-	
-	
+
+
 	PetscFunctionBeginUser;
 	ierr = DMDAGetInfo(veloc_da,0,&M,&P,NULL,0,0,0, 0,0,0,0,0,0);CHKERRQ(ierr);
-	
-	
+
+
 	ierr = VecZeroEntries(local_V);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(veloc_da,Veloc,INSERT_VALUES,local_V);
 	ierr = DMGlobalToLocalEnd(  veloc_da,Veloc,INSERT_VALUES,local_V);
-	
+
 	ierr = DMDAVecGetArray(veloc_da,local_V,&VV);CHKERRQ(ierr);
-	
-	
-	
+
+
+
 	/* get acces to the vector */
 
 	ierr = VecZeroEntries(local_FV);CHKERRQ(ierr);
 	ierr = DMDAVecGetArray(veloc_da,local_FV,&ff);CHKERRQ(ierr);
-	
+
 
 	ierr = VecZeroEntries(local_FP);CHKERRQ(ierr);
 	ierr = DMDAVecGetArray(veloc_da,local_FP,&ffp);CHKERRQ(ierr);
-	
-	
+
+
 	ierr = VecZeroEntries(local_dRho);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(drho_da,dRho,INSERT_VALUES,local_dRho);
 	ierr = DMGlobalToLocalEnd(  drho_da,dRho,INSERT_VALUES,local_dRho);
-	
+
 	ierr = DMDAVecGetArray(drho_da,local_dRho,&rr);CHKERRQ(ierr);
-	
-	
+
+
 	ierr = VecZeroEntries(local_P);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(veloc_da,Pressure,INSERT_VALUES,local_P);
 	ierr = DMGlobalToLocalEnd(  veloc_da,Pressure,INSERT_VALUES,local_P);
-	
+
 	ierr = DMDAVecGetArray(veloc_da,local_P,&pp);CHKERRQ(ierr);
-	
-	
+
+
 	Stokes					**VVC;
-	
+
 	ierr = VecZeroEntries(local_VC);CHKERRQ(ierr);
-	
+
 	ierr = DMGlobalToLocalBegin(veloc_da,Veloc_Cond,INSERT_VALUES,local_VC);
 	ierr = DMGlobalToLocalEnd(  veloc_da,Veloc_Cond,INSERT_VALUES,local_VC);
-	
+
 	ierr = DMDAVecGetArray(veloc_da,local_VC,&VVC);CHKERRQ(ierr);
 	//////
-	
-	
 
-	
-	
-	
+
+
+
+
+
 	PetscInt i,j,k,c,n,g;
-	
+
 	MatStencil indr[T_NE],ind[V_GT];
-	
+
 	ierr = DMDAGetElementCorners(drho_da,&sex,&sez,&mx,&mz);CHKERRQ(ierr);
-	
+
 	PetscInt               sex1,sez1,mx1,mz1;
 
 	ierr = DMDAGetElementCorners(veloc_da,&sex1,&sez1,&mx1,&mz1);CHKERRQ(ierr);
-	
+
 	if ((sex-sex1!=0)||(sez-sez1!=0)||(mx-mx1!=0)||(mz-mz1!=0)){
 		printf("%d %d %d %d\n",sex-sex1,sez-sez1,mx-mx1,mz-mz1);
 		SETERRQ1(PETSC_COMM_WORLD,1,"Wrong partition (temper,velocity)\n",1);
@@ -453,19 +452,19 @@ PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
 	for (ek = sez; ek < sez+mz; ek++) {
 		for (ei = sex; ei < sex+mx; ei++) {
 			//montaKeThermal_simplif(TCe, TKe, 0, 0);//modificar
-			
+
 			//for (c=0;c<T_NE*T_NE;c++) Ttotal_b[c] = TMe[c] - comp_alpha_thermal*dt_calor_sec*TCe[c];
-			
-			
+
+
 			indr[0].i=ei  ; indr[0].j=ek  ;
 			indr[1].i=ei+1; indr[1].j=ek  ;
 			indr[2].i=ei  ; indr[2].j=ek+1;
 			indr[3].i=ei+1; indr[3].j=ek+1;
-			
-			
+
+
 			for (i=0;i<T_NE;i++) dr[i] = rr[indr[i].j][indr[i].i];
-			
-			
+
+
 			for (i=0;i<V_GT;i++){
 				Vfe[i] = 0.0;
 				for (j=0;j<T_NE;j++){
@@ -473,57 +472,57 @@ PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
 				}
 				Vfe_P[i] = Vfe[i] - VCe[i]*pp[ek][ei].u;
 			}
-			
+
 			n=0;
 			ind[n].i=ei  ; ind[n].j=ek  ; ind[n].c=0; n++;
 			ind[n].i=ei  ; ind[n].j=ek  ; ind[n].c=1; n++;
-			
+
 			ind[n].i=ei+1; ind[n].j=ek  ; ind[n].c=0; n++;
 			ind[n].i=ei+1; ind[n].j=ek  ; ind[n].c=1; n++;
-			
+
 			ind[n].i=ei  ; ind[n].j=ek+1; ind[n].c=0; n++;
 			ind[n].i=ei  ; ind[n].j=ek+1; ind[n].c=1; n++;
-			
+
 			ind[n].i=ei+1; ind[n].j=ek+1; ind[n].c=0; n++;
 			ind[n].i=ei+1; ind[n].j=ek+1; ind[n].c=1; n++;
-			
-			
-			
-			
+
+
+
+
 			for (n=0;n<4;n++){
 				g = 2*n;
 				if (VVC[indr[n].j][indr[n].i].u==0){
 					Vfe[g]=0.0;
 					Vfe_P[g]=0.0;
 				}
-				
+
 				g = 2*n+1;
 				if (VVC[indr[n].j][indr[n].i].w==0){
 					Vfe[g]=0.0;
 					Vfe_P[g]=0.0;
 				}
 			}
-			
-			
+
+
 			for (c=0;c<V_GT;){
 				ff[ind[c].j][ind[c].i].u += Vfe[c]; c++;
 				ff[ind[c].j][ind[c].i].w += Vfe[c]; c++;
 			}
-			
+
 			for (c=0;c<V_GT;){
 				ffp[ind[c].j][ind[c].i].u += Vfe_P[c]; c++;
 				ffp[ind[c].j][ind[c].i].w += Vfe_P[c]; c++;
 			}
-			
-			
+
+
 		}
 	}
-	
-	
+
+
 	PetscInt       sx,sz,mmx,mmz;
-	
+
 	ierr = DMDAGetCorners(veloc_da,&sx,&sz,NULL,&mmx,&mmz,NULL);CHKERRQ(ierr);
-	
+
 	for (k=sz; k<sz+mmz; k++) {
 		for (i=sx; i<sx+mmx; i++) {
 			if (periodic_boundary==1 && i==Nx-1) ;
@@ -537,24 +536,24 @@ PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
 			}
 		}
 	}
-	
-	
+
+
 	ierr = DMDAVecRestoreArray(veloc_da,local_FV,&ff);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalBegin(veloc_da,local_FV,ADD_VALUES,F);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalEnd(veloc_da,local_FV,ADD_VALUES,F);CHKERRQ(ierr);
-	
+
 	ierr = DMDAVecRestoreArray(veloc_da,local_FP,&ffp);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalBegin(veloc_da,local_FP,ADD_VALUES,FP);CHKERRQ(ierr);
 	ierr = DMLocalToGlobalEnd(veloc_da,local_FP,ADD_VALUES,FP);CHKERRQ(ierr);
-	
+
 	ierr = DMDAVecRestoreArray(drho_da,local_dRho,&rr);CHKERRQ(ierr);
 	ierr = DMDAVecRestoreArray(veloc_da,local_P,&pp);CHKERRQ(ierr);
 	ierr = DMDAVecRestoreArray(veloc_da,local_VC,&VVC);CHKERRQ(ierr);
-	
+
 	ierr = DMDAVecRestoreArray(veloc_da,local_V,&VV);CHKERRQ(ierr);
 
 
-	
+
 	if (periodic_boundary==1){
 		ierr = DMGlobalToLocalBegin(veloc_da,FP,INSERT_VALUES,local_FP);
 		ierr = DMGlobalToLocalEnd(  veloc_da,FP,INSERT_VALUES,local_FP);
@@ -579,7 +578,7 @@ PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
 		ierr = DMDAVecRestoreArray(veloc_da,local_FV,&ff);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(veloc_da,local_FV,INSERT_VALUES,F);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(veloc_da,local_FV,INSERT_VALUES,F);CHKERRQ(ierr);
-		
+
 		ierr = DMDAVecRestoreArray(veloc_da,local_FP,&ffp);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(veloc_da,local_FP,INSERT_VALUES,FP);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(veloc_da,local_FP,INSERT_VALUES,FP);CHKERRQ(ierr);
@@ -606,91 +605,91 @@ PetscErrorCode AssembleF_Veloc(Vec F,DM veloc_da,DM drho_da,Vec FP){
 		ierr = DMDAVecRestoreArray(veloc_da,local_FV,&ff);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(veloc_da,local_FV,INSERT_VALUES,F);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(veloc_da,local_FV,INSERT_VALUES,F);CHKERRQ(ierr);
-		
+
 		ierr = DMDAVecRestoreArray(veloc_da,local_FP,&ffp);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(veloc_da,local_FP,INSERT_VALUES,FP);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(veloc_da,local_FP,INSERT_VALUES,FP);CHKERRQ(ierr);
 	}
-	
-	
+
+
 	PetscFunctionReturn(0);
 }
 
 PetscErrorCode Init_Veloc(){
-	
-	
+
+
 	PetscErrorCode         ierr;
-	
+
 	PetscMPIInt rank;
 	ierr = MPI_Comm_rank(PETSC_COMM_WORLD,&rank);CHKERRQ(ierr);
-	
-	
-	
+
+
+
 	PetscInt ix[1];
 	PetscScalar y[1];
-	
+
 	PetscInt low,high;
-	
+
 	PetscInt t;
 
-	
+
 	if (veloc_extern==1){
 		char s1[100],s2[100];
-		
+
 		sprintf(s1,"input_velocity_%d.txt",i_veloc);
 		sprintf(s2,"veloc_init.bin");
 
 		PetscPrintf(PETSC_COMM_WORLD,"Reading %s\n",s1);
-		
+
 		if (rank==0){
 			ierr = ascii2bin(s1,s2); CHKERRQ(ierr);
 		}
 		MPI_Barrier(PETSC_COMM_WORLD);
-		
-		
+
+
 		PetscInt size0;
 		PetscViewer    viewer;
-		
+
 		VecGetSize(Veloc,&size0);
-		
-		
+
+
 		Vec Fprov;
-		
-		
+
+
 		PetscViewerBinaryOpen(PETSC_COMM_WORLD,s2,FILE_MODE_READ,&viewer);
 		VecCreate(PETSC_COMM_WORLD,&Fprov);
 		VecLoad(Fprov,viewer);
 		PetscViewerDestroy(&viewer);
-		
-		
+
+
 		VecGetOwnershipRange(Fprov,&low,&high);
-		
-		
-		
+
+
+
 		Vec FN;
-		
+
 		DMDACreateNaturalVector(da_Veloc,&FN);
-		
-		
+
+
 		for (t=low;t<high;t++){
 			ix[0] = t;
 			VecGetValues(Fprov,1,ix,y);
 			VecSetValue(FN,t,y[0], INSERT_VALUES);
 		}
-		
+
 		VecAssemblyBegin(FN);
 		VecAssemblyEnd(FN);
-		
+
 		DMDANaturalToGlobalBegin(da_Veloc,FN,INSERT_VALUES,Veloc);
 		DMDANaturalToGlobalEnd(da_Veloc,FN,INSERT_VALUES,Veloc);
 
 		ierr = VecDestroy(&FN);CHKERRQ(ierr);
 		ierr = VecDestroy(&Fprov);CHKERRQ(ierr);
-		
-	
-		
+
+
+
 		PetscBarrier(NULL);
-		
+
 		VecAssemblyBegin(Veloc);
 		VecAssemblyEnd(Veloc);
 
@@ -698,33 +697,33 @@ PetscErrorCode Init_Veloc(){
 
 	}
 	else {
-		
+
 		Stokes					**VV;
-		
+
 		ierr = VecZeroEntries(local_V);CHKERRQ(ierr);
 		ierr = DMDAVecGetArray(da_Veloc,local_V,&VV);CHKERRQ(ierr);
-		
+
 		PetscInt       sx,sz,mmx,mmz;
-		
+
 		ierr = DMDAGetCorners(da_Veloc,&sx,&sz,NULL,&mmx,&mmz,NULL);CHKERRQ(ierr);
-		
+
 		PetscInt               M,P;
-		
+
 		ierr = DMDAGetInfo(da_Veloc,0,&M,&P,NULL,0,0,0, 0,0,0,0,0,0);CHKERRQ(ierr);
-		
+
 		ierr = DMDAVecRestoreArray(da_Veloc,local_V,&VV);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(da_Veloc,local_V,INSERT_VALUES,Veloc);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(da_Veloc,local_V,INSERT_VALUES,Veloc);CHKERRQ(ierr);
 	}
-	
-	
-	
+
+
+
 	VecCopy(Veloc,Veloc_fut);
 	VecCopy(Veloc,Veloc_0);
-	
+
 	write_veloc_3d(-1,binary_output);
 	write_veloc_cond(-1,binary_output);
-	
+
 	PetscFunctionReturn(0);
-	
+
 }

--- a/src/DMv.cpp
+++ b/src/DMv.cpp
@@ -151,20 +151,20 @@ PetscErrorCode create_veloc_2d(PetscInt mx,PetscInt mz,PetscInt Px,PetscInt Pz)
 	PetscInt       dof,stencil_width;
 
 	PetscErrorCode ierr;
-	
-	
-	
+
+
+
 	int rank;
 	MPI_Comm_rank(PETSC_COMM_WORLD,&rank);
-	
+
 	PetscLogDouble Tempo1p,Tempo2p;
-	
+
 	PetscTime(&Tempo1p);
-	
-	
+
+
 	PetscFunctionBeginUser;
-	
-	
+
+
 	dof           = 2; //check for the future 3d version
 	if (periodic_boundary==0){
 		stencil_width = 1;
@@ -178,24 +178,24 @@ PetscErrorCode create_veloc_2d(PetscInt mx,PetscInt mz,PetscInt Px,PetscInt Pz)
 	}
 	ierr = DMSetFromOptions(da_Veloc);CHKERRQ(ierr);
 	ierr = DMSetUp(da_Veloc);CHKERRQ(ierr);
-	
+
 	ierr = DMDASetFieldName(da_Veloc,0,"V_x");CHKERRQ(ierr);
 	ierr = DMDASetFieldName(da_Veloc,1,"V_z");CHKERRQ(ierr);
-	
-	
+
+
 	ierr = PetscCalloc1(V_GT*V_GT,&Ke_veloc); CHKERRQ(ierr);
 	ierr = PetscCalloc1(V_GT*V_GT,&Ke_veloc_final); CHKERRQ(ierr);
 
 	ierr = PetscCalloc1(V_GT*V_GT*GaussQuad,&Ke_veloc_general); CHKERRQ(ierr);
-	
+
 	ierr = PetscCalloc1(V_GT,&Vfe); CHKERRQ(ierr);
-	
-	ierr = PetscCalloc1(V_GT*V_NE,&VfMe); CHKERRQ(ierr); 
+
+	ierr = PetscCalloc1(V_GT*V_NE,&VfMe); CHKERRQ(ierr);
 	ierr = PetscCalloc1(V_GT,&VCe); CHKERRQ(ierr);
-	
-	
+
+
 	ierr = DMDASetUniformCoordinates(da_Veloc,0.0,Lx,-depth,0.0,-1,-1);CHKERRQ(ierr); //check: in 2D, the last two are ignored
-	
+
 
 
 	ierr = DMSetMatType(da_Veloc,MATAIJ);CHKERRQ(ierr);
@@ -205,113 +205,113 @@ PetscErrorCode create_veloc_2d(PetscInt mx,PetscInt mz,PetscInt Px,PetscInt Pz)
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_fut);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_weight);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_0);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_step1);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_step2);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&Vf);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&Vf_P);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&Pressure);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateMatrix(da_Veloc,&VG);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&rk_vec2);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&rk_vec);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&sk_vec);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&gs_vec);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&uk_vec);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&Adiag);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&zk_vec);CHKERRQ(ierr);
 	ierr = DMCreateGlobalVector(da_Veloc,&zk_vec2);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateGlobalVector(da_Veloc,&Precon);CHKERRQ(ierr);
 
 	ierr = DMCreateGlobalVector(da_Veloc,&Veloc_Cond);CHKERRQ(ierr);
-	
-	
+
+
 	ierr = DMCreateLocalVector(da_Veloc,&local_V);CHKERRQ(ierr);
 	ierr = DMCreateLocalVector(da_Veloc,&local_VC);CHKERRQ(ierr);
 	ierr = DMCreateLocalVector(da_Veloc,&local_FV);CHKERRQ(ierr);
 	ierr = DMCreateLocalVector(da_Veloc,&local_FP);CHKERRQ(ierr);
 	ierr = DMCreateLocalVector(da_Veloc,&local_P);CHKERRQ(ierr);
-	
+
 	ierr = DMCreateLocalVector(da_Veloc,&local_Precon);CHKERRQ(ierr);
-	
-	
+
+
 	Stokes					**ff;
 	PetscInt               M,P;
-	
+
 	ierr = DMDAGetInfo(da_Veloc,0,&M,&P,NULL,0,0,0, 0,0,0,0,0,0);CHKERRQ(ierr);
-	
-	
+
+
 	ierr = VecZeroEntries(local_FV);CHKERRQ(ierr);
 	ierr = DMDAVecGetArray(da_Veloc,local_FV,&ff);CHKERRQ(ierr);
-	
+
 	PetscInt       sx,sz,mmx,mmz;
 	PetscInt i,k,t;
-	
+
 	ierr = DMDAGetCorners(da_Veloc,&sx,&sz,NULL,&mmx,&mmz,NULL);CHKERRQ(ierr);
-	
+
 	PetscInt ix[1];
 	PetscScalar y[1];
-	
+
 	PetscInt low,high;
-	
+
 	if (bcv_extern==1){
 		char s1[100],s2[100];
-		
+
 		sprintf(s1,"bcv_0.txt");
 		sprintf(s2,"bcv_init.bin");
-		
+
 		if (rank==0){
 			ierr = ascii2bin(s1,s2); CHKERRQ(ierr);
 		}
 		MPI_Barrier(PETSC_COMM_WORLD);
-		
-		
+
+
 		PetscInt size0;
 		PetscViewer    viewer;
-		
+
 		VecGetSize(Veloc_Cond,&size0);
-		
-		
+
+
 		Vec Fprov;
-		
-		
+
+
 		PetscViewerBinaryOpen(PETSC_COMM_WORLD,s2,FILE_MODE_READ,&viewer);
 		VecCreate(PETSC_COMM_WORLD,&Fprov);
 		VecLoad(Fprov,viewer);
 		PetscViewerDestroy(&viewer);
-		
-		
+
+
 		VecGetOwnershipRange(Fprov,&low,&high);
-		
+
 		Vec FN;
-		
+
 		DMDACreateNaturalVector(da_Veloc,&FN);
-		
-		
+
+
 		for (t=low;t<high;t++){
 			ix[0] = t;
 			VecGetValues(Fprov,1,ix,y);
 			VecSetValue(FN,t,y[0], INSERT_VALUES);
 		}
-		
+
 		VecAssemblyBegin(FN);
 		VecAssemblyEnd(FN);
-		
+
 		DMDANaturalToGlobalBegin(da_Veloc,FN,INSERT_VALUES,Veloc_Cond);
 		DMDANaturalToGlobalEnd(da_Veloc,FN,INSERT_VALUES,Veloc_Cond);
-		
-		
+
+
 		//VecView(F,PETSC_VIEWER_STDOUT_WORLD);
-		
+
 		PetscBarrier(NULL);
-		
+
 		VecAssemblyBegin(Veloc_Cond);
 		VecAssemblyEnd(Veloc_Cond);
 
@@ -322,99 +322,99 @@ PetscErrorCode create_veloc_2d(PetscInt mx,PetscInt mz,PetscInt Px,PetscInt Pz)
 		VecView(Veloc_Cond,viewer2);
 		PetscViewerDestroy(&viewer2);
 		//VecView(F,PETSC_VIEWER_STDOUT_WORLD);
-		
+
 	}
 	else {
 		for (k=sz; k<sz+mmz; k++) {
 			for (i=sx; i<sx+mmx; i++) {
 				ff[k][i].u = 1.0;
 				ff[k][i].w = 1.0;
-				
+
 				if (periodic_boundary==0){
 					if (i==0   && bcv_left_normal==1) ff[k][i].u = 0.0;
 					if (i==0   && bcv_left_slip==1) {
 						ff[k][i].w = 0.0;
 					}
-				
+
 					if (i==M-1 && bcv_right_normal==1)ff[k][i].u = 0.0;
 					if (i==M-1   && bcv_right_slip==1) {
 						ff[k][i].w = 0.0;
 					}
 				}
-				
+
 				if (k==0   && bcv_bot_normal==1) ff[k][i].w = 0.0;
 				if (k==0   && bcv_bot_slip==1){
 					ff[k][i].u = 0.0;
 				}
-				
+
 				if (k==P-1 && bcv_top_normal==1) ff[k][i].w = 0.0;
 				if (k==P-1 && bcv_top_slip==1){
 					ff[k][i].u = 0.0;
 				}
-				
+
 			}
 		}
 		ierr = DMDAVecRestoreArray(da_Veloc,local_FV,&ff);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalBegin(da_Veloc,local_FV,INSERT_VALUES,Veloc_Cond);CHKERRQ(ierr);
 		ierr = DMLocalToGlobalEnd(da_Veloc,local_FV,INSERT_VALUES,Veloc_Cond);CHKERRQ(ierr);
 	}
-	
-	
-	
+
+
+
 	//if (rank==0) printf("Init_veloc\n");
 	Init_Veloc();
 	//if (rank==0) printf("Init_veloc: fim\n");
-	
-	
+
+
 	int ind;
 	PetscReal r;
-	
+
 	VecMax(Veloc_Cond,&ind,&r);
-	
+
 	//if (rank==0) printf("VecMax Veloc_Cond: %f\n",r);
-	
+
 	VecSum(Veloc_Cond,&r);
-	
+
 	//if (rank==0) printf("VecSum Veloc_Cond: %f\n",r);
-	
-	
-	
+
+
+
 	ierr = KSPCreate(PETSC_COMM_WORLD,&V_ksp);CHKERRQ(ierr);
   ierr = KSPSetDM(V_ksp,da_Veloc);CHKERRQ(ierr);
   ierr = KSPSetDMActive(V_ksp,PETSC_FALSE);CHKERRQ(ierr);
 	ierr = KSPSetOptionsPrefix(V_ksp,"veloc_"); CHKERRQ(ierr);
-	
+
 	PetscTime(&Tempo2p);
 	if (rank==0) printf("Velocity field (creation): %lf s\n",Tempo2p-Tempo1p);
-	
+
 	montaKeVeloc_general(Ke_veloc_general,dx_const,dz_const);
-	
+
 	montaCeVeloc(VCe);
 	montafeVeloc(VfMe);
-	
-	
+
+
 	PetscFunctionReturn(0);
-	 
+
 }
 
 PetscErrorCode build_veloc_3d()
 {
-	
+
 	PetscErrorCode ierr;
-	
+
 	int rank;
 	MPI_Comm_rank(PETSC_COMM_WORLD,&rank);
-	
+
 	PetscLogDouble Tempo1p,Tempo2p;
-	
+
 	PetscTime(&Tempo1p);
-	
+
 	ierr = MatZeroEntries(VA);CHKERRQ(ierr);
-	
+
 	ierr = MatZeroEntries(VB);CHKERRQ(ierr);
 	ierr = VecZeroEntries(Vf);CHKERRQ(ierr);
 	ierr = VecZeroEntries(Vf_P);CHKERRQ(ierr);
-	
+
 	ierr = VecZeroEntries(Precon);CHKERRQ(ierr);
 	if (Verif_first_veloc==1)	VecCopy(Veloc_fut,Veloc_weight);
 	Verif_first_veloc=1;
@@ -428,23 +428,23 @@ PetscErrorCode build_veloc_3d()
 
 	ierr = moveSwarm(0.0);
 	ierr = Swarm2Mesh();
-	
-	
+
+
 	ierr = AssembleA_Veloc(VA,VG,da_Veloc,da_Thermal);CHKERRQ(ierr);
-	
+
 	ierr = VecReciprocal(Precon);
-	
+
 	ierr = calc_drho();CHKERRQ(ierr);
-	
-	
+
+
 	ierr = AssembleF_Veloc(Vf,da_Veloc,da_Thermal,Vf_P);CHKERRQ(ierr);
-	
-	
+
+
 	PetscTime(&Tempo2p);
 	if (rank==0) printf("  Velocity field (building): %lf s\n",Tempo2p-Tempo1p);
-	
+
 	PetscFunctionReturn(0);
-	
+
 }
 
 
@@ -456,17 +456,17 @@ PetscErrorCode solve_veloc_3d()
 	PetscLogDouble Tempo1,Tempo2;
 
 	PC V_pc;
-	
+
 	int rank;
-	
-	
+
+
 	MPI_Comm_rank(PETSC_COMM_WORLD,&rank);
-	
+
 	PetscTime(&Tempo1);
-	
-	
+
+
 	/* SOLVE */
-	
+
 	//if (rank==0) printf("k\n");
 	ierr = KSPSetOperators(V_ksp,VA,VA);CHKERRQ(ierr);
 	if (direct_solver==1){
@@ -479,107 +479,107 @@ PetscErrorCode solve_veloc_3d()
 	ierr = KSPSetFromOptions(V_ksp);CHKERRQ(ierr);
 	//if (rank==0) printf("k\n");
 	ierr = KSPSetInitialGuessNonzero(V_ksp,PETSC_TRUE);
-	
+
 	ierr = KSPSetTolerances(V_ksp,rtol,PETSC_DEFAULT,PETSC_DEFAULT,PETSC_DEFAULT);CHKERRQ(ierr);
-	
-	
+
+
 	////////
-	
+
 	PetscReal denok=0,betak,alphak;
-	
+
 	PetscInt maxk = 400,k;
-	
+
 	ierr = KSPSolve(V_ksp,Vf_P,Veloc_fut);CHKERRQ(ierr); /// Solve KV0 = F
-	
+
 	PetscInt its;
-	
+
 	ierr = KSPGetIterationNumber(V_ksp,&its);CHKERRQ(ierr);
-	
+
 	VecPointwiseMult(Veloc_fut,Veloc_fut,Veloc_Cond); ///check: zero values at the b.c.
 	VecAXPY(Veloc_fut,1.0,Veloc_0); ///check: applying Veloc_0 in Veloc at the b.c.
-	
+
 	//write_veloc_3d(101);
-	
-	
+
+
 	ierr = MatMultTranspose(VG,Veloc_fut,rk_vec2);CHKERRQ(ierr); /// r0 = G^T V0
-	
-	
+
+
 	ierr = VecPointwiseMult(zk_vec2,Precon,rk_vec2); /// z0 = Precon*r0 = M^-1 *r0 <----
-	
+
 	ierr = VecDot(rk_vec2,rk_vec2,&denok);CHKERRQ(ierr); //denok = r0^2 ?
-	
+
 	if (rank==0) printf("    Uzawa iteration: 0, denok = %lg, its = %d\n",denok, its);
-	
-	
-	
+
+
+
 	for (k=1;k<maxk && denok>denok_min;k++){ /// while denok>denok_min
 		if (k==1) VecCopy(zk_vec2,sk_vec); ///if k=1: s1 = zk <----
 		else {
 			VecCopy(zk_vec2,zk_vec);
-			
+
 			ierr = VecPointwiseMult(zk_vec2,Precon,rk_vec2); /// z_(k-1) = Precon*r_(k-1) <----
-			
+
 			VecDot(zk_vec2,rk_vec2,&betak);
 			VecDot(zk_vec,rk_vec,&denok);
 			betak=betak/denok; /// bk = (z_(k-1)*r_(k-1))/(z_(k-2)*r_(k-2)) <----
 			VecAYPX(sk_vec,betak,zk_vec2); /// sk = z_(k-1) + b*s_(k-1) <----
 		}
 		ierr = MatMult(VG,sk_vec,gs_vec);
-		
+
 		VecPointwiseMult(gs_vec,gs_vec,Veloc_Cond);
-		
+
 		VecDot(gs_vec,gs_vec,&denok);
-		
-		
-		
+
+
+
 		KSPSolve(V_ksp,gs_vec,uk_vec); /// K uk = G sk
 		VecPointwiseMult(uk_vec,uk_vec,Veloc_Cond);//zero values at the b.c.
-		
+
 		ierr = KSPGetIterationNumber(V_ksp,&its);CHKERRQ(ierr);
-		
+
 		VecDot(zk_vec2,rk_vec2,&alphak);
 		VecDot(gs_vec,uk_vec,&denok);
-		
+
 		alphak=alphak/denok; /// ak = (r_(k-1) z_(k-1))/((G sk)uk) <----
-		
+
 		VecAXPY(Pressure,alphak,sk_vec); /// Pk = P_(k-1) + ak*sk
-		
+
 		VecAXPY(Veloc_fut,-alphak,uk_vec); /// Vk = V_(k-1) - ak*uk
-		
+
 		VecCopy(rk_vec2,rk_vec);
-		
+
 		ierr = MatMultTranspose(VG,uk_vec,rk_vec2);CHKERRQ(ierr);
-		
+
 		VecAYPX(rk_vec2,-alphak,rk_vec); ///rk = r_(k-1) - ak*(G uk)
-		
+
 		VecDot(rk_vec2,rk_vec2,&denok);
-		
+
 		if (rank==0) printf("    Uzawa iteration: %d, denok = %lg, its = %d\n",k,denok,its);
-		
+
 	}
 
 	shift_pressure();
-	
+
 	////////
-	
-	
+
+
 	PetscTime(&Tempo2);
 	if (rank==0) printf("    Velocity field (solution): %lf s\n",Tempo2-Tempo1);
-	
+
 	PetscFunctionReturn(0);
-	
+
 }
 
 
 PetscErrorCode destroy_veloc_3d()
 {
-	
+
 	int rank;
-	
+
 	MPI_Comm_rank(PETSC_COMM_WORLD,&rank);
 	PetscLogDouble Tempo1,Tempo2;
 	PetscTime(&Tempo1);
-	
+
 	PetscErrorCode ierr;
 	ierr = KSPDestroy(&V_ksp);CHKERRQ(ierr);
 	ierr = VecDestroy(&Veloc_weight);CHKERRQ(ierr);
@@ -590,11 +590,11 @@ PetscErrorCode destroy_veloc_3d()
 	ierr = MatDestroy(&VA);CHKERRQ(ierr);
 	ierr = MatDestroy(&VB);CHKERRQ(ierr);
 	ierr = DMDestroy(&da_Veloc);CHKERRQ(ierr);
-	
+
 	PetscTime(&Tempo2);
 	if (rank==0) printf("Velocity field (destroying): %lf s\n",Tempo2-Tempo1);
-	
-	
+
+
 	PetscFunctionReturn(0);
 }
 
@@ -613,7 +613,7 @@ PetscErrorCode write_veloc_cond(int cont, PetscInt binary_out)
 
 	sprintf(variable_name,"bc_velocity");
 	write_all_(cont,Veloc_Cond,variable_name,binary_out);
-	
+
 	PetscFunctionReturn(0);
 }
 
@@ -627,32 +627,32 @@ PetscErrorCode write_veloc_cond(int cont, PetscInt binary_out)
 
 
 PetscErrorCode montaKeVeloc_general(PetscReal *KeG, double dx_const, double dz_const){
-	
+
 	long i,j,ii;
 	long aux;
-	
-	
-	
+
+
+
 	double kx,kz;
-	
+
 	double ex,ez;
-	
+
 	long cont;
-	
+
 	double N_x[V_NE];
 	double N_z[V_NE];
-	
+
 	double SN[3][V_GT];
-	
+
 	for (i=0;i<3;i++){
 		for (j=0;j<V_GT;j++){
 			SN[i][j]=0;
 		}
 	}
-	
-	
+
+
 	long point;
-	
+
 	for (point=0;point<GaussQuad;point++){
 		for (i=0;i<V_GT;i++){
 			for (j=0;j<V_GT;j++){
@@ -660,10 +660,10 @@ PetscErrorCode montaKeVeloc_general(PetscReal *KeG, double dx_const, double dz_c
 			}
 		}
 	}
-	
+
 	double Hx,Hz,prodH;
-	
-	
+
+
 	point=0;
 	for (kz=-r06; kz<=r06; kz+=r06){
 		if (kz==0) Hz=r8p9;
@@ -678,25 +678,25 @@ PetscErrorCode montaKeVeloc_general(PetscReal *KeG, double dx_const, double dz_c
 					//N[cont]=(1+ex*kx)*(1+ez*kz)/4.0;
 					N_x[cont]=ex*(1+ez*kz)/2.0/dx_const; //check: only valid in 2D
 					N_z[cont]=(1+ex*kx)*ez/2.0/dz_const; //check: only valid in 2D
-					
+
 					cont++;
 				}
 			}
-			
-			
+
+
 			for (j=0;j<V_NE;j++){
 				aux = j*V_GN;
 				SN[0][aux  ]=N_x[j];
 				SN[1][aux+1]=N_z[j];
-				
+
 				SN[2][aux  ]=N_z[j];SN[2][aux+1]=N_x[j];
-				
+
 				//Nd[aux  ]=N_x[j];
 				//Nd[aux+1]=N_y[j];
 				//Nd[aux+2]=N_z[j];
 			}
-			
-			
+
+
 			for (i=0;i<V_GT;i++){
 				for (j=0;j<V_GT;j++){
 					for (ii=0;ii<2;ii++){
@@ -705,60 +705,55 @@ PetscErrorCode montaKeVeloc_general(PetscReal *KeG, double dx_const, double dz_c
 					for (;ii<3;ii++){
 						KeG[(i*V_GT+j)+point*V_GT*V_GT]+=prodH*SN[ii][i]*SN[ii][j];
 					}
-					
+
 				}
 			}
-			
+
 			point++;
-			
+
 		}
-	
+
 	}
-	
+
 	/*for (i=0;i<V_NE*GaussQuad;i++){
 		PetscPrintf(PETSC_COMM_WORLD,"%lg %lg %lg\n",N_x_Gauss[i],N_y_Gauss[i],N_z_Gauss[i]);
 	}
 	exit(1);*/
-	
-	
+
+
 	PetscFunctionReturn(0);
-	
+
 }
 
 
 
 PetscReal montaKeVeloc_simplif(PetscReal *Ke,PetscReal *KeG, PetscReal *geoq_ele){
-	
+
 	long i,j;
-	
+
 	double Visc_local,Geoq_local;
 	double visc_meio=-1.0;
-	
+
 	double kx,kz;
-	
+
 	double ex,ez;
-	
+
 	long cont;
-	
-	
-	double Hx,Hz;
-	
+
+
+
 	long point=0;
-	
+
 	for (i=0;i<V_GT*V_GT;i++) Ke[i]=0.0;
-	
+
 	if (visc_const_per_element==0){
-		
+
 		for (kz=-r06; kz<=r06; kz+=r06){
-			if (kz==0) Hz=r8p9;
-			else Hz=r5p9;
-			
+
 			for (kx=-r06; kx<=r06; kx+=r06){
-				if (kx==0) Hx=r8p9;
-				else Hx=r5p9;
-				
+
 				Geoq_local = 0.0;
-				
+
 				cont=0;
 				for (ez=-1.;ez<=1.;ez+=2.){
 					for (ex=-1.;ex<=1.;ex+=2.){
@@ -766,69 +761,64 @@ PetscReal montaKeVeloc_simplif(PetscReal *Ke,PetscReal *KeG, PetscReal *geoq_ele
 						cont++;
 					}
 				}
-				
+
 				Visc_local = Geoq_local;
-				
+
 				if (kx==0 && kz==0) visc_meio = Visc_local;
-				
-				
+
+
 				if (Visc_local<visc_aux_MIN) visc_aux_MIN=Visc_local;
 				if (Visc_local>visc_aux_MAX) visc_aux_MAX=Visc_local;
-				
-				
+
+
 				for (i=0;i<V_GT;i++){
 					for (j=0;j<V_GT;j++){
 						Ke[i*V_GT+j]+=KeG[(i*V_GT+j)+point]*Visc_local;
 					}
 				}
-				
+
 				point+=V_GT*V_GT;
 			}
-		
+
 		}
 	}
 	else {
 		double Visc_mean;
 
 		Geoq_local = geoq_ele[0];
-				
+
 		Visc_local = Geoq_local;
-		
+
 		Visc_mean = Visc_local;
-		
-		
+
+
 		if (Visc_local<visc_aux_MIN) visc_aux_MIN=Visc_local;
 		if (Visc_local>visc_aux_MAX) visc_aux_MAX=Visc_local;
-		
+
 		for (kz=-r06; kz<=r06; kz+=r06){
-			if (kz==0) Hz=r8p9;
-			else Hz=r5p9;
-			
+
 			for (kx=-r06; kx<=r06; kx+=r06){
-				if (kx==0) Hx=r8p9;
-				else Hx=r5p9;
-				
-				
+
 				for (i=0;i<V_GT;i++){
 					for (j=0;j<V_GT;j++){
 						Ke[i*V_GT+j]+=KeG[(i*V_GT+j)+point];
 					}
 				}
-				
+
 				point+=V_GT*V_GT;
 			}
-		
+
 		}
 		visc_meio = Visc_mean;
-		
+
 		for (i=0;i<V_GT;i++){
 			for (j=0;j<V_GT;j++){
 				Ke[i*V_GT+j]*=Visc_mean;
 			}
 		}
 	}
-	
-	
+
+
 	return(visc_meio);
 
 }
@@ -837,31 +827,31 @@ PetscReal montaKeVeloc_simplif(PetscReal *Ke,PetscReal *KeG, PetscReal *geoq_ele
 PetscErrorCode montafeVeloc(PetscReal *fMe)
 {
 	long i,j;
-	
+
 	double kx,kz;
-	
+
 	double ex,ez;
-	
+
 	long cont;
-	
+
 	double N[V_NE];
-	
+
 	for (i=0;i<V_GT;i++){
 		for (j=0;j<V_NE;j++){
 			fMe[i*V_NE+j]=0.0;
 		}
 	}
-	
+
 	double Hx,Hz,prodH;
 	for (kz=-r06; kz<=r06; kz+=r06){
 		if (kz==0) Hz=r8p9;
 		else Hz=r5p9;
-		
+
 		for (kx=-r06; kx<=r06; kx+=r06){
 			if (kx==0) Hx=r8p9;
 			else Hx=r5p9;
-			
-			
+
+
 			prodH = Hx*Hz;
 			cont=0;
 			for (ez=-1.;ez<=1.;ez+=2.){
@@ -870,56 +860,56 @@ PetscErrorCode montafeVeloc(PetscReal *fMe)
 					cont++;
 				}
 			}
-			
-			
+
+
 			for (i=0;i<V_NE;i++){
 				for (j=0;j<V_NE;j++){
 					fMe[(i*V_GN+1)*V_NE+j]+=prodH*N[i]*N[j]; ///check: in 2d it is +1
 				}
 			}
-			
+
 		}
-	
+
 	}
-	
+
 	for (i=0;i<V_GT*V_NE;i++){
 		fMe[i]*=dx_const*dz_const;
 	}
 
-	
+
 	PetscFunctionReturn(0);
 }
 
 
 PetscErrorCode montaCeVeloc(PetscReal *Ce){
-	
+
 	long i;
-	
-	
+
+
 	double kx,kz;
-	
+
 	double ex,ez;
-	
+
 	long cont;
-	
+
 	double N_x[V_NE];
 	double N_z[V_NE];
-	
+
 	for (i=0;i<V_GT;i++){
 		Ce[i]=0.0;
 	}
-	
+
 	double Hx,Hz,prodH;
-	
+
 	for (kz=-r06; kz<=r06; kz+=r06){
 		if (kz==0) Hz=r8p9;
 		else Hz=r5p9;
-		
+
 		for (kx=-r06; kx<=r06; kx+=r06){
 			if (kx==0) Hx=r8p9;
 			else Hx=r5p9;
-			
-			
+
+
 			prodH = Hx*Hz;
 			cont=0;
 			for (ez=-1.;ez<=1.;ez+=2.){
@@ -930,21 +920,21 @@ PetscErrorCode montaCeVeloc(PetscReal *Ce){
 					cont++;
 				}
 			}
-			
+
 			for (i=0;i<V_NE;i++){
 				Ce[i*V_GN]+=prodH*N_x[i]; // 1/8 shape function for pressure (constant in the element) in 3D
 				Ce[i*V_GN+1]+=prodH*N_z[i];
 			}
-			
+
 		}
-		
+
 	}
-	
+
 	for (i=0;i<V_GT;i++){
 		Ce[i]*=-dx_const*dz_const;
 	}
-	
+
 	PetscFunctionReturn(0);
 
-	
+
 }

